lib/livebook_web/plugs/auth_plug.ex

```diff
@@ -16,36 +16,63 @@ defmodule LivebookWeb.AuthPlug do
   def init(opts), do: opts
 
   @impl true
-  def call(conn, _otps) do
-    case Application.get_env(:livebook, :token) do
-      nil -> conn
-      token -> token_authentication(conn, token)
+  def call(conn, opts) do
+    cond do
+      Application.fetch_env!(:livebook, :password_authentication) == true ->
+        password = Application.fetch_env!(:livebook, :password)
+        secret_authentication(conn, password, :password, opts)
```

Comment:
I am not sure it is a good practice to store the password in the cookie. The token is fine because it is random and it changes on every new instance. The password can be persisted. My suggestion is to not unify the token and password approaches. Instead, have the authentication store the information in the session and in here you check instead if the user is in the session.

Reply:
Yeah, I think it's gonna be cleaner to have separate function for each approach, there's probably not gonna be that much duplication anyway. As for keeping the session, one option I see is storing a hashed version of the password in the session, as this shouldn't pose too much of a risk, while still allows for comparing if the authentication was done against the same password, WDYT?

Sidenote: with a single config variable for auth mode we can nicely `case` match here :D

Reply:
I took @jonatanklosko 's approach of hashing the password with sha256 but when I think right now about it I suppose that it would be easier to brute force password combinations than to brute force cookie decryption...